tests/unit/s2n_resume_test.c

```diff
@@ -0,0 +1,248 @@
+            EXPECT_SUCCESS(s2n_stuffer_init(&secret_stuffer, &secret));
+            EXPECT_SUCCESS(s2n_stuffer_write_bytes(&secret_stuffer, test_master_secret.data, S2N_TLS_SECRET_LEN));
+            conn->secure.cipher_suite = &s2n_ecdhe_ecdsa_with_aes_128_gcm_sha256;
+
+            uint8_t data[S2N_TICKET_SIZE_IN_BYTES] = { 0 };
+            struct s2n_blob blob = { 0 };
+            struct s2n_stuffer output = { 0 };
+            EXPECT_SUCCESS(s2n_blob_init(&blob, data, sizeof(data)));
+            EXPECT_SUCCESS(s2n_stuffer_init(&output, &blob));
+
+            EXPECT_SUCCESS(s2n_encrypt_session_ticket(conn, NULL, &output));
+
+            /* Wiping the master secret to prove that the decryption function actually writes the master secret */
+            memset(conn->secure.master_secret, 0, test_master_secret.size);
+
+            conn->client_ticket_to_decrypt = output;
+            EXPECT_SUCCESS(s2n_decrypt_session_ticket(conn));
+
+            /* Check decryption was successful by comparing master key */
+            EXPECT_BYTEARRAY_EQUAL(conn->secure.master_secret, test_master_secret.data, test_master_secret.size);
+
+            EXPECT_SUCCESS(s2n_connection_free(conn));
+            EXPECT_SUCCESS(s2n_config_free(config));
+        }
+
+        /* Check session ticket size is correct for TLS13. The contents of the output will be
+         * tested once the TLS1.3 deserialization function is written. */
+        {
+            struct s2n_connection *conn;
+            struct s2n_config *config;
+            uint64_t current_time;
+            EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_SERVER));
+            EXPECT_NOT_NULL(config = s2n_config_new());
+
+            EXPECT_SUCCESS(s2n_config_set_session_tickets_onoff(config, 1));
+            EXPECT_SUCCESS(config->wall_clock(config->sys_clock_ctx, &current_time));
+            EXPECT_SUCCESS(s2n_config_add_ticket_crypto_key(config, ticket_key_name, strlen((char *)ticket_key_name),
+                         ticket_key.data, sizeof(ticket_key), current_time/ONE_SEC_IN_NANOS));
+
+            EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+
+            conn->actual_protocol_version = S2N_TLS13;
+
+            uint8_t data[S2N_TICKET_KEY_NAME_LEN + S2N_TLS_GCM_IV_LEN + S2N_MAX_STATE_SIZE_IN_BYTES + S2N_TLS_GCM_TAG_LEN] = { 0 };
+            struct s2n_blob blob = { 0 };
+            struct s2n_stuffer output = { 0 };
+            EXPECT_SUCCESS(s2n_blob_init(&blob, data, sizeof(data)));
+            EXPECT_SUCCESS(s2n_stuffer_init(&output, &blob));
+            struct s2n_ticket_fields ticket_fields = { .ticket_age_add = 1, .session_secret = test_session_secret };
+
+            EXPECT_SUCCESS(s2n_encrypt_session_ticket(conn, &ticket_fields, &output));
+
+            uint32_t expected_size = S2N_TICKET_KEY_NAME_LEN + S2N_TLS_GCM_IV_LEN + 
+                        S2N_TLS13_STATE_SIZE_WITHOUT_SECRET + test_session_secret.size + S2N_TLS_GCM_TAG_LEN;
```

Comment:
To be very sure this is testing what you want to test, we should probably assert that test_session_secret.size is not the maximum TLS1.3 secret size. If it is, then your test working might just be a coincidence.

You might actually want two versions of this test: one for a smaller secret, one for the maximum size secret.